src/components/implementation/no_interface/scheddev/sl_lock.c

```diff
@@ -2,109 +2,67 @@
 
 #include "sl.h"
 
-void sl_lock_init(struct sl_lock *lock) {
+void
+sl_lock_init(struct sl_lock *lock)
+{
     assert(lock);
-
-    *lock = (struct sl_lock) {
-        .holder = 0
-    };
+    *lock = SL_LOCK_STATIC_INIT();
 }
 
-thdid_t sl_lock_holder(struct sl_lock *lock) {
+thdid_t
+sl_lock_holder(struct sl_lock *lock)
+{
     assert(lock);
     return lock->holder;
 }
 
-// If timeout is 0, ignore the timeout
-int sl_lock_timeout_elapsed(microsec_t start_time, microsec_t timeout) {
-    if (timeout == 0 || (sl_now_usec() - start_time) <= timeout) {
-        return 0;
-    }
-    return 1;
-}
 
-int sl_lock_lock_internal_no_cs(struct sl_lock *lock, microsec_t timeout) {
+void
+sl_lock_lock(struct sl_lock *lock)
+{
     assert(lock);
 
-    microsec_t start_time = sl_now_usec();
-    while (lock->holder != 0 && !sl_lock_timeout_elapsed(start_time, timeout)) {
-        thdid_t holder = lock->holder;
-
-        /*
-         * If we are preempted after the exit, and the holder is no longer holding
-         * the critical section, then we will yield to them and possibly waste a
-         * time-slice.  This will be fixed the next iteration, as we will see an
-         * updated value of the holder, but we essentially lose a timeslice in the
-         * worst case.  From a real-time perspective, this is bad, but we're erring
-         * on simplicity here.
-         */
-        sl_cs_exit();
-        sl_thd_yield(holder);
+    sl_cs_enter();
+    while (lock->holder != 0) {
+        sl_thd_yield_cs_exit(lock->holder);
         sl_cs_enter();
     }
-
```

Comment:
I like my current code better than the first one. The second version has a subtle race, where we get preempted after reading the holder, and the lock as released. Then we waste a timeslice yielding to no one.